### PR TITLE
Rename Swagger Modules to OpenApi

### PR DESCRIPTION
**Pull request by DimalChandrasiri**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -3,10 +3,10 @@
...
 
 ## Flexible
-You can build your integrations from sequence diagrams, or write code in the Ballerina language or Swagger. You can also use plug-ins to write Ballerina code in IntelliJ IDEA, Vim, and more. Ballerina is open source, so you can see exactly how it works and join the growing community of users contributing to its code base. And its API allows you to leverage its functionality from other programs.
+You can build your integrations from sequence diagrams, or write code in the Ballerina language or in Open API Specification. You can also use plug-ins to write Ballerina code in IntelliJ IDEA, Vim, and more. Ballerina is open source, so you can see exactly how it works and join the growing community of users contributing to its code base. And its API allows you to leverage its functionality from other programs.
```
Comment (NipunaMarcus) [inline docs/index.md]: OpenAPI is a single name ... Lets remove the space. 

What if we say "using OpenAPI Specification"

File: docs/index.md [hunk 2]
```diff
@@ -3,10 +3,10 @@
...
 
 ## Powerful
-The Ballerina language was designed from the ground up specifically for integration and can handle everything from a simple Hello World program to complex service chaining and content-based routing scenarios. It comes with native support for REST, Swagger, JSON, and XML, and it includes connectors for popular services like Twitter and Facebook. The runtime is lightweight and incredibly fast, and you can run it alone without the development tools in a production environment.  
+The Ballerina language was designed from the ground up specifically for integration and can handle everything from a simple Hello World program to complex service chaining and content-based routing scenarios. It comes with native support for REST, Open API, JSON, and XML, and it includes connectors for popular services like Twitter and Facebook. The runtime is lightweight and incredibly fast, and you can run it alone without the development tools in a production environment.
```
Comment (NipunaMarcus) [inline docs/index.md]: OpenAPI is a single name. There shouldn't be any spaces in between

File: misc/openapi-ballerina/modules/ballerina-client-generator/spotbugs-exclude.xml [hunk 3]
```diff
@@ -15,5 +15,5 @@
   ~ specific language governing permissions and limitations
   ~ under the License.
   -->
-<FindBugsFilter>
-</FindBugsFilter>
\ No newline at end of file
+<findbugsfilter>
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/spotbugs-exclude.xml]: I think this element's name is "FindBugsFilter" .. can you please check?

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java [hunk 4]
```diff
@@ -40,20 +40,20 @@
 
     /**
      * Generates code(client, server etc) for ballerina source provided  in <code>definitionPath</code>.
-     * <p>Method can be used for generating OAS 3.0.0, Swagger or Ballerina Client</p>
+     * <p>Method can be used for generating OAS 3.0.0, OpenApi or Ballerina Client</p>
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java]: OpenAPI is mentioned twice here by using "OAS 3.0.0" and "OpenAPI"

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java [hunk 5]
```diff
@@ -40,20 +40,20 @@
...
      *
      * @param type    Output type. Following types are supported
      *                <ul>
      *                <li>client</li>
      *                <li>openapi</li>
-     *                <li>swagger</li>
+     *                <li>openapi</li>
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java]: here also openapi is mentioned twice.

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java [hunk 6]
```diff
@@ -63,8 +63,8 @@ public static String generateOutput(GeneratorConstants.GenType type, ClientConte
                 return getConvertedString(context, GeneratorConstants.DEFAULT_OPEN_API_DIR,
                         GeneratorConstants.OPEN_API_TEMPLATE_NAME);
             case SWAGGER:
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java]: Need to change this constant's name "SWAGGER" to "OPENAPI"

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/GeneratorConstants.java [hunk 7]
```diff
@@ -40,18 +40,18 @@ public String toString() {
...
-    public static final String SWAGGER_TEMPLATE_NAME = "skeleton";
+    public static final String OPENAPI_TEMPLATE_NAME = "skeleton";
 
     public static final String TEMPLATES_SUFFIX = ".mustache";
     public static final String TEMPLATES_DIR_PATH_KEY = "templates.dir.path";
     public static final String DEFAULT_TEMPLATE_DIR = "/templates";
     public static final String DEFAULT_CLIENT_DIR = DEFAULT_TEMPLATE_DIR + "/client";
     public static final String DEFAULT_OPEN_API_DIR = DEFAULT_TEMPLATE_DIR + "/oas3";
-    public static final String DEFAULT_SWAGGER_DIR = DEFAULT_TEMPLATE_DIR + "/swagger2";
+    public static final String DEFAULT_OPENAPI_DIR = DEFAULT_TEMPLATE_DIR + "/swagger2";
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/GeneratorConstants.java]: we need to change the name of this dir... lets asses the impact

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/ConverterConstants.java [hunk 8]
```diff
@@ -14,14 +14,14 @@
...
+package org.ballerinalang.ballerina.openapi.convertor.service;
 
 /**
- * This class will hold all constants related to swagger ballerina conversion.
+ * This class will hold all constants related to openapi ballerina conversion.
  */
 public class ConverterConstants {
     public static final String YAML_EXTENSION = ".yaml";
-    public static final String SWAGGER_SUFFIX = ".swagger";
+    public static final String OPENAPI_SUFFIX = ".swagger";
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/ConverterConstants.java]: we have to check the .swagger extension and is there any changes to it.

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java [hunk 9]
```diff
@@ -76,7 +76,7 @@
      * @param httpAlias    The alias for ballerina/http module.
      * @param swaggerAlias The alias for ballerina.swagger module.
      */
-    SwaggerResourceMapper(Swagger swagger, String httpAlias, String swaggerAlias) {
+    OpenApiResourceMapper(Swagger swagger, String httpAlias, String swaggerAlias) {
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java]: can we change the parameter names to openApi

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java [hunk 10]
```diff
@@ -64,7 +64,7 @@
 /**
  * This class will do resource mapping from ballerina to swagger.
  */
-public class SwaggerResourceMapper {
+public class OpenApiResourceMapper {
     private final String httpAlias;
     private final String swaggerAlias;
     private final Swagger swaggerDefinition;
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java]: can we change the "swaggerDefinition" and "swaggerAlias" ?

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiServiceMapper.java [hunk 11]
```diff
@@ -88,38 +88,38 @@ public String generateSwaggerString(Swagger swagger) {
      * This method will convert ballerina @Service to swaggers @Swagger object.
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiServiceMapper.java]: is there a possibility to change "swagger" in doc comments also?

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiServiceMapper.java [hunk 12]
```diff
@@ -88,38 +88,38 @@ public String generateSwaggerString(Swagger swagger) {
...
-        SwaggerResourceMapper resourceMapper = new SwaggerResourceMapper(swagger, this.httpAlias, this.swaggerAlias);
-        swagger.setPaths(resourceMapper.convertResourceToPath(service.getResources()));
-        return swagger;
+        OpenApiResourceMapper resourceMapper = new OpenApiResourceMapper(openapi, this.httpAlias, this.swaggerAlias);
+        openapi.setPaths(resourceMapper.convertResourceToPath(service.getResources()));
+        return openapi;
     }
 
     /**
      * Parses the 'ServiceInfo' annotation and build the swagger definition for it.
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiServiceMapper.java]: change "swagger" in doc comment

File: misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/cmd/OpenApiCmd.java [hunk 13]
```diff
@@ -35,9 +35,9 @@
  * Ex: ballerina swagger (mock | client) (swaggerFile | balFile) -m(module name) -o(output directory name)
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/cmd/OpenApiCmd.java]: remember to change after the CLI command change

File: docs/index.md (hunk 1 above)
Comment (DimalChandrasiri) [reply docs/index.md]: Resloved with 81d9190

File: docs/index.md (hunk 2 above)
Comment (DimalChandrasiri) [reply docs/index.md]: Resolved with 81d9190

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java (hunk 4 above)
Comment (DimalChandrasiri) [reply misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java]: Resolved with 81d9190

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java (hunk 5 above)
Comment (DimalChandrasiri) [reply misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java]: Resolved with 81d9190

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java (hunk 9 above)
Comment (DimalChandrasiri) [reply misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java]: Resolved with 81d9190

File: misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java (hunk 10 above)
Comment (DimalChandrasiri) [reply misc/openapi-ballerina/modules/ballerina-to-openapi-generator/src/main/java/org/ballerinalang/ballerina/openapi/convertor/service/OpenApiResourceMapper.java]: Resolved with 81d9190

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/GeneratorConstants.java (hunk 7 above)
Comment (DimalChandrasiri) [reply misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/GeneratorConstants.java]: This is a temp location which is only used in the OpenApi Client Generator to generate the temp OAS file. Hence IMO will be able to change the location name.

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java [hunk 14]
```diff
@@ -40,31 +40,30 @@
...
-            case OPENAPI:
+            case OPENAPI3:
                 return getConvertedString(context, GeneratorConstants.DEFAULT_OPEN_API_DIR,
                         GeneratorConstants.OPEN_API_TEMPLATE_NAME);
-            case SWAGGER:
-                return getConvertedString(context, GeneratorConstants.DEFAULT_SWAGGER_DIR,
-                        GeneratorConstants.SWAGGER_TEMPLATE_NAME);
+            case OPENAPI:
+                return getConvertedString(context, GeneratorConstants.DEFAULT_OPENAPI_DIR,
+                        GeneratorConstants.OPENAPI_TEMPLATE_NAME);
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/CodeGenerator.java]: OPENAPI_TEMPLATE_NAME and OPEN_API_TEMPLATE_NAME should be named after the version of the spec

File: misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/GeneratorConstants.java [hunk 15]
```diff
@@ -26,7 +26,7 @@
      *
      */
     public enum GenType {
-        CLIENT("client"), OPENAPI("oas3.0"), SWAGGER("swagger");
+        CLIENT("client"), OPENAPI3("oas3.0"), OPENAPI("swagger");
```
Comment (NipunaMarcus) [inline misc/openapi-ballerina/modules/ballerina-client-generator/src/main/java/org/ballerinalang/client/generator/GeneratorConstants.java]: change swagger to openAPI2

**Pull request merged**